README.md

```diff
@@ -162,7 +222,24 @@ The ST-LINK debugger can then be connected via header CN4 (pins 1-5 with 1 neare
 
 After deploying the binary to the board, the Nucleo's `LD1` LED will flash red and green. Programming is complete when the LED stays green, so don't powercycle the board before this.
 
-## Serial Communication
+## Serial Wire Output (SWO) (Rev 2 +)
+
+The 10-pin Serial Wire Debug (SWD) programming interface does not come with UART lines for standard printf usage. Instead, we will use Serial Wire Output (SWO), a single wire interface to transmit trace messages to an external debugger.
+
+See `apps/test-logger` for an example of using the SWO-supported logger utility. Ensure that the target you are building for is configured to support SWO logging (see targets/<**target name**>/include/mbed_config_target.h).
+
+### Ubuntu
+
+**TODO**
+
```

Comment:
i think we should figure this out before merging

Reply:
actually if you've tried and cant find a solution there's no need to keep trying. ig its important but not sar critical since aidan has windows

Reply:
I think this can be made an issue for after SAR